execute/plugin_functions.go

```diff
@@ -154,6 +154,31 @@ func filterOutExecutedMessages(
 
 	var filtered []plugintypes.ExecutePluginCommitData
 
+	// ----------------------------------------
+	// TODO: optimize algorithm
+	executedSeqNums := mapset.NewSet[cciptypes.SeqNum]()
+	for _, execedRange := range executedMessages {
+		for seqNum := execedRange.Start(); seqNum <= execedRange.End(); seqNum++ {
+			executedSeqNums.Add(seqNum)
+		}
+	}
+
+	// mark executed messages for each report
+	for reportIdx := range reports {
+		report := &reports[reportIdx]
+		executedSeqNumsForReport := mapset.NewSet[cciptypes.SeqNum]()
+		for seqNum := report.SequenceNumberRange.Start(); seqNum <= report.SequenceNumberRange.End(); seqNum++ {
+			if executedSeqNums.Contains(seqNum) {
+				executedSeqNumsForReport.Add(seqNum)
+			}
+		}
+		report.ExecutedMessages = executedSeqNumsForReport.ToSlice()
+		sort.Slice(report.ExecutedMessages, func(i, j int) bool {
+			return report.ExecutedMessages[i] < report.ExecutedMessages[j]
+		})
+	}
+	// ----------------------------------------
+
```

Comment:
We need a test case for this. I'm not sure what this does differently than the original logic

Reply:
not needed eventually, removing it

Reply:
was fine before